src/components/views/elements/PersistentApp.tsx

```diff
@@ -88,17 +111,34 @@ export default class PersistentApp extends React.Component<IProps, IState> {
     };
 
     public render(): JSX.Element {
-        if (this.state.persistentWidgetId) {
-            const persistentWidgetInRoomId = ActiveWidgetStore.instance.getRoomId(this.state.persistentWidgetId);
+        const wId = this.state.persistentWidgetId;
+        if (wId) {
+            const persistentWidgetInRoomId = ActiveWidgetStore.instance.getRoomId(wId);
 
             const persistentWidgetInRoom = MatrixClientPeg.get().getRoom(persistentWidgetInRoomId);
 
             // Sanity check the room - the widget may have been destroyed between render cycles, and
             // thus no room is associated anymore.
             if (!persistentWidgetInRoom) return null;
 
-            const myMembership = persistentWidgetInRoom.getMyMembership();
-            if (this.state.roomId !== persistentWidgetInRoomId && myMembership === "join") {
+            const wls = WidgetLayoutStore.instance;
+
+            const userIsPartOfTheRoom = persistentWidgetInRoom.getMyMembership() == "join";
+            const fromAnotherRoom = this.state.roomId !== persistentWidgetInRoomId;
+
+            const notInRightPanel =
+                !(this.state.rightPanelPhase == RightPanelPhases.Widget &&
+                wId == RightPanelStore.getSharedInstance().roomPanelPhaseParams?.widgetId);
+            const notInCenterContainer =
+                    !wls.getContainerWidgets(persistentWidgetInRoom, Container.Center)
+                        .find((app)=>app.id == wId);
+            const notInTopContainer =
+                !wls.getContainerWidgets(persistentWidgetInRoom, Container.Top).find(app=>app.id == wId);
+            if (
+                //Show the persistent widget in two cases. The booleans have to be read like this: the widget is-`fromAnotherRoom`:
```

Comment:
Reading the booleans is not really the concern for this block of code, but description of the two cases is. Can the comment be updated to mention the product rationale for why the conditions are the way they are?